src/Gui/ReportView.cpp

```diff
@@ -470,6 +491,10 @@ void ReportOutput::onToggleError()
     getWindowParameter()->SetBool( "checkError", bErr );
 }
 
+void ReportOutput::onToggleShowReportViewOnWarningOrError(){

```

Comment:
You can completely get rid of the bShow member in this case. Just write:
bool show = getWindowParameter()->GetBool("checkShowReportViewOnWarningOrError", true);
getWindowParameter()->SetBool("checkShowReportViewOnWarningOrError", !show);